src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,328 @@
+    
+    #Load CBSD
+    device_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_4 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_d.json')))
+
+    # Load GWPZ Record
+    gwpz_record_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+
+    # Load FSS record
+    fss_record_1 =  json.load(
+        open(os.path.join('testcases', 'testdata', 'fss_record_0.json')))
+    fss_record_2 =  json.load(
+        open(os.path.join('testcases', 'testdata', 'fss_record_1.json')))
+
+    # Load ESC record
+    esc_record_1 =  json.load(
+        open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+
+
+    # Create the actual config.
+    config = {
+        'cbsdRecords': [{'registrationRequests':[device_1,device_2],'grantRequests':[grant_record_1,grant_record_2]},
+                        {'registrationRequests':[device_3,device_4],'grantRequests':[grant_record_3,grant_record_4]}],
+        'IAPRecords': [[
+      { 'type': 'FSS','Record': [fss_record_1] },
+      { 'type': 'ESC','Record': [esc_record_1] } ],
+      [ { 'type': 'GWPZ','Record': [gwpz_record_1]},
+      { 'type': 'FSS', 'Record': [fss_record_2]}]],
+	'sas_hareness': [{'name': 'SAS-TH-1',
+			'url': ['localhost', '9001'],
+			'cert_file': 'certs/server_a.cert',
+			'key_file': 'certs/server_a.key'},
+	                {
+			'name': 'SAS-TH-2',
+			'url': ['localhost', '9002'],
+			'cert_file': 'certs/server_b.cert',
+			'key_file': 'certs/server_b.key'
+		        }],
+        'sas_uut': {'name': 'SAS-UUT',
+                        'url': ['localhost', '9000'],
+                        'cert_file': 'certs/server.cert',
+                        'key_file': 'certs/server.key'}, 
+        'K':3, 
+        'headroom':{'MgPpa':2,'MgGwpz':2,'MgCochannel':2,'MgBlocking':2,'MgOobe':2,'MgEsc':2},
+        'DPAactiveInfo':[],
+        'DPAdeactiveInfo':[]
+       }
+    
+    writeConfig(filename, config)
+
+
+  @configurable_testcase(generate_MCP_1_default_config) 
+  def test_WINNF_FT_S_MCP_1(self,config_filename):
+    """SAS manages a mix of GAA and PAL Grants in 3550 MHz
+       to 3700 MHz to protect configurable IAP-protected entities and DPAs
+    """
+    config = loadConfig(config_filename)
+    pre_iap_headroom = config ['headroom']
+
+    
+    #Step 1 : Load DPAs
+    self._sas_admin.TriggerLoadDpas()
+
+    #STEP 2 : ESC informs SAS about inactive DPA
+    self._sas_admin.TriggerBulkDpaActivation({'activate':False})
+
+    #Step 3 : creates multiple SAS TH, and Load predefined FAD,CBSD 
+    self.createSASTH(config)
+    self.configureFadForEachSASTH(0)
+    
+    #fad objects get the fad dumps from the remote SAS configured with 'url'
```

Comment:
It looks like this is actually fetching from the local SAS Test Harnesses, no?

Reply:
here we   are configuring the SAS Test harness with FAD objects as per the step 3 in   test case